backend/src/main/java/com/google/step/servlets/WebhookServlet.java

```diff
@@ -59,15 +66,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       response.sendRedirect("/");
       return;
     }
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Query query = new Query("Lead").addSort("date", SortDirection.DESCENDING);
-    PreparedQuery queryResults = datastore.prepare(query);
-    ArrayList<Lead> leads = new ArrayList<>();
-    for (Entity leadEntity : queryResults.asIterable()) {
-      leads.add(new Lead(leadEntity));
+    User user = UserAuthenticationUtil.getCurrentUser();
+    //generate URL-Safe Key string
+    String advertiserKeyString = KeyFactory.keyToString(AdvertiserUtil.createAdvertiserKey(user));
+    String webhookUrl = request.getScheme() + "://" +
+            request.getServerName() + ":" +
+            request.getServerPort() + "/api/webhook?id=" +
+            advertiserKeyString;
+    //generate random google key (currently 20 chars)
+    Random rand = new SecureRandom();
+    StringBuilder googleKeyBuilder = new StringBuilder(20);
+    for (int i = 0; i < 20; i++) {
+      googleKeyBuilder.append(alphanumerics[rand.nextInt(alphanumerics.length)]);
```

Comment:
This does not ensure uniqueness. Do you think this will be a problem?

Reply:
Yeah, will make sure its unique after I implement the Form object to storage the google keys (since haven't stored them yet).

Reply:
Decided google keys don't need to be unique for now. Will keep this until further notice.